common/db.py

```diff
@@ -349,6 +347,7 @@ def update_finalized_batches(self, app_name: str, sig_data: dict[str, Any]) -> N
         target_batch: dict[str, Any] = batches[sig_data["hash"]]
         target_batch["finalization_signature"] = sig_data["signature"]
         target_batch["finalized_nonsigners"] = sig_data["nonsigners"]
+        target_batch["tag"] = sig_data["tag"]
```

Comment:
it should be "finalized_tag" just like "finalized_nonsigners" to not conflict with the locking tag if they differ.